content/en/docs/Tracing/tempo-walkthrough.md

```diff
@@ -0,0 +1,206 @@
+All other components setup should not be kind specific but may require different configuration depending on the target cluster.
+
+## Cluster Setup
+
+In this first step we are going to create a local cluster using [kind](https://kind.sigs.k8s.io).
+
+The created cluster will have two nodes, one master node and one worker node.
+Note that the master node maps host ports `80` and `443` to the container node, if those ports are already in use they can be changed by editing the `hostPort` stanza in the config manifest below.
+
+To create the local cluster run the following command:
+
+```shell
+kind create cluster --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+    kubeadmConfigPatches:
+      - |-
+        kind: InitConfiguration
+        nodeRegistration:
+          kubeletExtraArgs:
+            node-labels: "ingress-ready=true"
+    extraPortMappings:
+      - containerPort: 80
+        hostPort: 80
+        protocol: TCP
+      - containerPort: 443
+        hostPort: 443
+        protocol: TCP
+  - role: worker
+EOF
+```
+
+## Ingress NGINX Setup
+
+In order to access Grafana from our browser, we need to deploy an ingress controller.
+
+We are going to install `ingress-nginx` with the following command:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+sleep 15
+kubectl wait --namespace ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=90s
+```
+
+## Grafana Setup
+
+Grafana will allow us to explore, search and examine traces.
+
+We can deploy Grafana using Helm with the following command:
+
+```shell
+helm upgrade --install grafana --namespace monitoring --create-namespace --wait \
+  --repo https://grafana.github.io/helm-charts grafana \
+  --values - <<EOF
+adminPassword: admin
+sidecar:
+  enableUniqueFilenames: true
+  dashboards:
+    enabled: true
+    searchNamespace: ALL
+    provider:
+      foldersFromFilesStructure: true
+  datasources:
+    enabled: true
+    searchNamespace: ALL
+grafana.ini:
+  server:
+    root_url: "%(protocol)s://%(domain)s:%(http_port)s/grafana"
+    serve_from_sub_path: true
+ingress:
+  enabled: true
+  path: /grafana
+  hosts: []
+EOF
+```
+
+At this point Grafana should be available at http://localhost/grafana (log in with `admin` / `admin`).
+
+## Tempo Setup
+
+Tempo is a tracing backend capable of receiving traces in OpenTelemetry format, it is developped and maintained by the Grafana team and integrates very well with it.
+
+We can deploy Tempo using Helm with the following command:
+
+```shell
+helm upgrade --install tempo --namespace monitoring --create-namespace --wait \
+  --repo https://grafana.github.io/helm-charts tempo \
+  --values - <<EOF
+tempo:
+  searchEnabled: true
+EOF
+```
+
+To make Tempo available in Grafana, we need to register it as a Grafana data source:
```

Comment:
Is it better to install this ConfigMap prior to creating Tempo so a restart of the deployment isn't necessary? If that's not the case, I don't see that step anywhere in here.

Reply:
I walked through the tutorial from scratch and didn't need to restart the pods.